File: ArchivosDePrueba/DesviacionEstandar.rb
def calcular_media(datos)
    suma = datos.reduce(0, :+)
    return suma / datos.length.to_f
  end
  
  def calcular_desviacion_estandar(datos)
    media = calcular_media(datos)
    suma_cuadrados_diferencia = datos.reduce(0) { |sum, x| sum + (x - media) ** 2 }
    varianza = suma_cuadrados_diferencia / datos.length.to_f
    return Math.sqrt(varianza)
  end
  
  # Ejemplo de uso
  conjunto_de_datos = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
  
  desviacion_estandar = calcular_desviacion_estandar(conjunto_de_datos)
  
  puts "Conjunto de datos: #{conjunto_de_datos}"
  puts "Desviación estándar: #{desviacion_estandar}"
  

File: ArchivosDePrueba/DesviacionEstandar.jl
# Datos de ejemplo
data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

# Calcular la media
mean_value = sum(data) / length(data)

# Calcular la suma de los cuadrados de las diferencias
sum_squared_diff = sum((x - mean_value)^2 for x in data)

# Calcular la varianza
variance = sum_squared_diff / (length(data) - 1)

# Calcular la desviación estándar como la raíz cuadrada de la varianza
std_deviation = sqrt(variance)

println("Desviación estándar: ", std_deviation)
